kaggle_environments/envs/werewolf/werewolf.py

```diff
@@ -0,0 +1,471 @@
+    # Register all specific LLM models. The AgentFactoryWrapper ensures each
+    # player gets a unique, stateful agent instance.
+    **{
+        f"llm/{model_name}": AgentFactoryWrapper(
+            LLMWerewolfAgent,
+            model_name=model_name,
+            system_prompt=LLM_SYSTEM_PROMPT
+        )
+        for model_name in LLM_MODEL_NAMES
+    }
+}
+
+
+def interpreter(state, env):
+    """
+    * Required interface function for kaggle environments package *
+
+    This is the primary interface for the kaggle environment (kEnv) to step game forward.
+    Briefly flow of logic is:
+    Initialization - kEnv creates werewolf object and chooses players. Schema definition for
+    this is in werewolf.json
+    1) kEnv calls interpreter() with current game state recorded in env.game_state 
+    2) interpreter() reads game state and any new player actions and updates
+       the games state based on those actions and flow of the game to env.game_state.
+    3) interpreter() writes events to history data and also writes events about 
+       state change in the game to env.game_state and returns back to kEnv
+    4) kEnv parses out the relevant game events via agent logic in harness/base.py, 
+       constructs final prompt, and performs external API calls for models and records back 
+       to env.game_state
+    Go back to 1 and continue
+
+    For example - consider discussion and voting by villagers. werewolf.interpreter()
+    updates phase and writes history entry that solicits players for discussion. 
+    kEnv calls agents to get their discussion and writes them to the history/game state.
+    kEnv then calls interpreter() that then updates game phase and writes history entry soliciting
+    votes for exile. kEnv then calls agents and associated models to get their votes and writes
+    responses to game state. env then calls interpreter() and moderator collects votes, determine
+    who was exiled, performs that action and advances game phase and game state.
+    And so on...
+
+    Note - The UI is also updated after each call to interpreter() as that is the tick unit
+    for the game.
+
+    Note - env framework assumes that there is an action to be done by player, but 
+    for werewolf there are places where moderator is the one taking the action (e.g
+    counting votes and performing exile) so some game 'ticks' are larger than others.
+
+    state: list of dictionaries, one for each agent.
+           Each dict has: {observation, action, reward, status, info}
+    env:   the kaggle_environments.Environment object itself including the env.game_state
+    """
+    # --- Initialize Moderator and GameState if it's the start of an episode ---
+    if not hasattr(env, 'moderator') or env.done:  # env.done is true after reset by Kaggle core
+        initialize_moderator(state, env)
+
+    moderator: Moderator = env.moderator
+    game_state: GameState = env.game_state
+
+    # 1. Collect and parse actions from Kaggle agents
```

Comment:
you'll need to handle `ERROR` and `TIMEOUT` states from players here

Reply:
(usually just mark everyone as DONE and end the episode as a tie, or that is what we did for chess)